meta/autoid/autoid.go

```diff
@@ -1062,46 +1100,55 @@ func TestModifyBaseAndEndInjection(alloc Allocator, base, end int64) {
 	alloc.(*allocator).mu.Unlock()
 }
 
-// ShardIDLayout is used to calculate the bits length of different segments in auto id.
-// Generally, an auto id is consist of 3 segments: sign bit, shard bits and incremental bits.
+// ShardIDFormat is used to calculate the bit length of different segments in auto id.
+// Generally, an auto id is consist of 4 segments: sign bit, reserved bits, shard bits and incremental bits.
 // Take ``a BIGINT AUTO_INCREMENT PRIMARY KEY`` as an example, assume that the `shard_row_id_bits` = 5,
 // the layout is like
-//  | [sign_bit] (1 bit) | [shard_bits] (5 bits) | [incremental_bits] (64-1-5=58 bits) |
-// Please always use NewShardIDLayout() to instantiate.
-type ShardIDLayout struct {
+//  | [sign_bit] (1 bit) | [reserved bits] (0 bits) | [shard_bits] (5 bits) | [incremental_bits] (64-1-5=58 bits) |
+// Please always use NewShardIDFormat() to instantiate.
+type ShardIDFormat struct {
 	FieldType *types.FieldType
 	ShardBits uint64
```

Comment:
question about type. Why they are int in `AutoRandomOption` and uint64 in `ShardIDFormat`?

Reply:
Because `AutoRandomOption` uses `UnspecifiedLength`(-1) as the default value.